engine/access/rpc/client/grpc_secure_client.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	libp2ptls "github.com/libp2p/go-libp2p-tls"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/onflow/flow-go/crypto"
+	"github.com/onflow/flow-go/network/p2p"
+)
+
+func main() {
```

Comment:
The following is how the client works:
1. From the remote node's public key hex string generate a Flow Crypto Public Networking key (our crypto format).
2. Convert the Flow crypto networking key to a Libp2p Crypto public networking key.
3. Generate a Libp2p Peer ID from the Libp2p Crypto networking key
4. Create a TLSConfig by using for this peer which does the validation that the public key received from the remote during a TLS handshake matches what we got by calling - `remote.MatchesPublicKey`

Reply:
Step 1 won't be needed in a collection node since it will already have the remote node public key as part of the remote flow.Identity